### DISCUSSION TITLE
Tell AFE what voltage range is selected in the Scoppy app?

### DISCUSSION THREAD

**Discussion by gilesp1729** [fhdm-dev/scoppy#56] [Scoppy Front end designs]
I know that you can get the AFE to inform Scoppy which of the 4 voltage ranges is selected via the two input pins. Would it be possible to do this the other way around - provide some output pins that tell  the AFE what voltage range has been selected on the display? That way software could control voltage range selection in the AFE (via CMOS switches or whatever) and no mechanical knobs/switches would be needed.

I'll look at the Pico source to see myself, but obviously you know the code better than me :-)

**Comment by fhdm-dev** [comment]
Yep. You can get it to do that but you'll need to compile the firmware with AUTO_VOLTAGE_RANGE defined.

The easiest way to do this is copy [scoppy-local-example.cmake](https://github.com/fhdm-dev/scoppy-pico/blob/master/pico/scoppy-local-example.cmake) to scoppy-local.cmake and uncomment the appropriate lines. Then just run cmake and make as you normally would.

You might also want to have a look at the documentation about INPUT_VOLTAGE_RANGES in scoppy-local-example.cmake.

Good luck :smile:

**Reply by gilesp1729** [reply]
Thanks for your prompt response!

**Reply by fhdm-dev** [reply]
Just thought I should let you know about a bug I found in the AUTO_VOLTAGE_RANGE functionality that could be a bit annoying if/when you create your AFE. If the Pico has connected to the Android device and then the app is closed and re-opened, the app will sometimes not recognise that AUTO_VOLTAGE_RANGE has been enabled in the firmware and the voltage range stuff just won't work. A workaround is to reset the Pico. There should be fix available for this in the next update.
